presto-hive/src/main/java/com/facebook/presto/hive/CachingDirectoryLister.java

```diff
@@ -78,9 +78,13 @@ public Iterator<HiveFileInfo> list(
             NamenodeStats namenodeStats,
             HiveDirectoryContext hiveDirectoryContext)
     {
-        List<HiveFileInfo> files = cache.getIfPresent(path);
-        if (files != null) {
-            return files.iterator();
+        if (hiveDirectoryContext.isCacheable()) {
```

Comment:
Thanks for fixing this, not having the check was resulting in incorrect result when the session property was set to disable the cache.